waku/v2/discv5/discover.go

```diff
@@ -197,74 +192,24 @@ func (d *DiscoveryV5) SetHost(h host.Host) {
 	d.host = h
 }
 
-type peerChannel struct {
```

Comment:
As i had mentioned above, it would be a better approach to extract this outside of discv5 and reuse it across all discovery services and this way we keep the other protocols(relay, filter etc) untouched.